h2o-core/src/test/java/water/runner/CleanNewKeysTask.java

```diff
@@ -0,0 +1,26 @@
+package water.runner;
+
+import water.*;
+
+import java.util.Set;
+
+public class CleanNewKeysTask extends MRTask<CleanNewKeysTask> {
+
+    @Override
+    protected void setupLocal() {
+        final Set<Key> initKeys = LocalTestRuntime.initKeys;
+        final Set<Key> actualKeys = H2O.localKeySet();
+        for (Key actualKey : actualKeys){
+            final Value keyValue = Value.STORE_get(actualKey);
+            if(initKeys.contains(actualKey) || isIgnorableKeyLeak(actualKey, keyValue)) continue;
+            H2O.STORE.remove(actualKey);
+            actualKey.remove();
+        }
+        
+    }
+
+    private static boolean isIgnorableKeyLeak(final Key key, final Value keyValue) {
```

Comment:
This function here and the one in CheckKeysTask looks exactly the same.  Wonder if you can combine them.

Reply:
Yes, this is doable.

Reply:
Done.